browser/src/control/Control.JSDialog.js

```diff
@@ -525,12 +532,33 @@ L.Control.JSDialog = L.Control.extend({
 		instance.posy = top + offsetY;
 
 		var width = instance.form.getBoundingClientRect().width;
-		var height = instance.form.getBoundingClientRect().height;
+		var autoFilterBottom = instance.posy + document.querySelector('.main-nav').clientHeight + document.getElementById('toolbar-wrapper').clientHeight + instance.form.getBoundingClientRect().height;
+		var toolbarTop = document.getElementById('spreadsheet-toolbar').getBoundingClientRect().top;
 		if (instance.posx + width > window.innerWidth)
 			instance.posx = window.innerWidth - width;
+
+		// at this point we have un updated potion of autofilter instance.
+		// so to handle overlapping case of autofiler and toolbar we need some complex calculation
+		if (autoFilterBottom > toolbarTop)
+			instance.posy = instance.posy - (autoFilterBottom - toolbarTop);
+
+		this.updatePosition(instance.container, instance.posx, instance.posy);
+	},
+
+	calculateSubmenuAutoFilterPosition: function(instance) {
+		// Get the position of the last focused element
+		var parentAutofilterEl = instance.lastFocusedElement.parentElement.getBoundingClientRect();
+		instance.posx = parentAutofilterEl.right + 20;
+		instance.posy = parentAutofilterEl.top - document.getElementById('toolbar-wrapper').offsetHeight;
+
+		// set posx of instance (submenufilter) based on window width 
+		if (instance.posx + instance.content.clientWidth > window.innerWidth)
```

Comment:
repeated `instance.content.clientWidth`, please use variable `width`. Also will be shorter, more easy to read.